include/nbl/builtin/hlsl/shapes/aabb.hlsl

```diff
@@ -17,32 +17,32 @@ namespace shapes
 struct AABB_t
 {
     //
-    void addPoint(const float3 pt)
+    void addPoint(const float32_t3 pt)
```

Comment:
can you template (except for the beziers) the shapes on a `scalar_t` then we can swap the type used to `float16_t` or `int16_t`

if you want a `float32_t3` then you simply do
```cpp
using vec3_t = vector<scalar_t,3>;
```
and use that typedef throughout the class